lib/xcpretty/parser.rb

```diff
@@ -193,7 +193,7 @@ module Warnings
 
       # @regex Captured groups
       # $1 = whole warning
-      GENERIC_WARNING_MATCHER = /^warning:\s(.*)$/
```

Comment:
To explain better myself - we should be defining `^` when a regex starts, so it skips the line as soon as possible.